### PR TITLE
Lavaland loot overhaul

### PR DESCRIPTION
**Pull request by Rukofamicom**
### This PR aims to reduce the maximum power ceiling for Shaft Miners and also increase the consistency of specific drops from bosses and tendrils.

***

Every boss now has a single unique item drop which they will always drop upon defeat, as well as a randomized cache of 100 ores (about 1200-1800 mining points worth, does not include bluespace crystals but does include diamond)
* Ash Drake always drops vial of dragon blood
* Colossus always drops the vocal chords
* Bubblegum always drops H.E.C.K. Suit
* Legion always drops an anomalous crystal (used to be part of Colossus drops)
* Hierophant always drops its staff (No change beyond now also dropping ore)
* Blood Drunk Miner always drops cleaving saw (Removed pka from its drop)

***

Most items useful only for pvp or grief have been removed from boss and necropolis chest drops both, and a few items widely considered to be "worthless" drops have been removed as well to compensate the overall quality of loot somewhat. Several of this items are equivalent to high-cost traitor items, or even so powerful there is no equivalent outside of Space Wizards.

Removed from Necro Chests:
* All three ERT-level Hardsuits
* Both Chaplain weapons
* Katana
* Infinite soulstone
* Mysterious Core
* Inferno Grenade
* Voodoo Doll
* Instant Summons Spellbook
* Puzzle Cube

Removed from Boss drop tables:
* **Spectral Blade** - Extremely annoying for ghosts due to being a permanent tether, and frequently a worthless drop beyond annoying them. When it wasn't worthless it's because the weapon had no upper damage cap
* **Lava staff** - Useful almost exclusively for grief considering the availability of RCDs. Even antagonists will have a somewhat hard time making use of the lava ability. 
* **Spellbook of Sacred Flame** - Again, almost exclusively useful for grief and nothing else.
* **Staff of Storms** - Ditto the above two items. 
* **Wand of Fireball** - Useless outside of PvP, and in PvP against anyone that isn't magic proof it's a one-hit ranged AoE stun and a 2-hit crit. 
* **Mayhem in a Bottle** - Useful for grief/murderbone and pretty much nothing else. If I tried to pass this as an uplink item, it would fail; so why is it available to non-antags just because they can kill a boss?
* **Blood Contract** - Who even thought this was a good idea? Same as above - this wouldn't even pass for a traitor uplink item and it's dedicated to the remote killing of someone. 
* **Spellblade** - While this one is substantially better balanced on damage, it's still exclusively useful for pvp and extraordinarily powerful for it due to the high dismember chance and duo of being powerful at both short and long range. 

***

Three of the remaining weapons have been rebalanced:
* **Hierophant Staff** has had its absurdly high 30 burn damage reduced to 15 across the board, and melee attacks now do 5 brute damage + 15 burn on AoE for a total of 20 damage. The staff remains one of the most powerful utility tools available due to its teleportation and ability to rapidly clear rocks as a mining tool, but it is no longer *also* an absurdly powerful weapon. 
* **Cleaving Saw** has had its damage against players reduced by 25% when unfolded and 33% when folded. It's bonus damage to fauna has been increased to compensate when using the unfolded form, but the bleeding effect while it is folded is so powerful that no tweaking was necessary to achieve the same number of hits until death. This is still an extremely powerful pvp weapon, and also still the highest DPS weapon for killing fauna as well, it's just been dialed back a little bit against players. 
* **Meat Hook** is now usable as a melee weapon when using harm intent... however its ranged damage has been decreased from 25 to 10. Melee swings deal 15 damage and follow normal armor calculations unlike the ranged attack. While this is a weapon that is useful almost exclusively for pvp, it's not *just* a weapon, and has its own unique gimmick while not being off the scales when it comes to power.

## Why It's Good For The Game

Members of security and antagonists should generally be the most powerfully equipped forces on the station when it comes to dealing with other players and rewards for mining should be heavily weighted toward rewards that are either directly related to mining or gimmicky fun - not items dedicated to grief and pvp.

Additionally many players wish to be miners just to learn how to fight the bosses and get better at fighting them - this allows players who find normal mining boring to still acquire ores that help the station as a whole as a reward for completing boss fights. No longer will miners rushing fauna be an issue to both the station and other miners who feel it isn't fair that they lose out because they decided to contribute to the station first. *If they choose to completely ignore the ore drops, it still won't help the station, but the players that do this can eat my shoe* 

## Changelog
:cl:
balance: Meat Hook Grab now does 10 Armor-piercing damage instead of 25. Can now be used for normal melee attacks as well.
balance: Hierophant club now does 20 damage on melee attack w/ AoE (down from 30), and 15 on homing projectile (down from 30)
balance: Open cleaving saw now does 15 (down from 20) damage to players and 60 (up from 50) damage to fauna
balance: Closed cleaving saw now does 8 (down from 12) damage, bonus bleed effect on fauna is unchanged, and this still kills all non-boss fauna in the same number of hits as before
del: Eleven items have been removed from Necropolis chest drop table: All three hardsuits, Chaplain weapons, Infinite Soulstone, Katana,  Mysterious Core, Inferno Grenade, Voodoo Doll, and the Instant Summons spellbook. 
del: Spectral Blade, Lava staff, Spellbook of Sacred Flame, Wand of Fireball, Mayhem in a Bottle, Blood Contract, Spellblade, and Staff of Storms have all been removed from megafauna drop tables. 
tweak: PKA-related rewards have increased drop chances for necropolis chests.
tweak: Lavaland puzzle chest now always awards PKA-related rewards
tweak: All megafauna now drop a moderately large caches of ores, roughly 1500 points worth. 
tweak: All megafauna will now always drop the same equipment reward every time they are killed, 
tweak: Anomalous crystals are now dropped by the greater Legion instead of Colossus.
tweak: Bottle of dragon's blood no longer has a chance to grant an Ash Drake transformation, but retained its other three effects.
/:cl:

### CODE REVIEW

**Comment by USB3pt0** [comment]
Lavaland loot nerf*

**Comment by utilbird** [comment]
To repost my thoughts from development:
> There's a lot of stuff being removed/rebalanced that wasn't even a problem
I'm all for reducing the ability or outright lootable removal of powergame pvp gear like katanas, ERT hardsuits and fireball wand
But there's a lot of goof stuff being removed that wasn't really much of a problem. (Hivelords are an exception, they're somewhere in the middle)
A good lavaland loot overhaul would be a part of a full on lavaland rework. It's a reward incentived job, making the rewards underwhelming will likely just lead to it becoming like the rest of the jobs in cargo department

> I don't like mining or it's rewards at all right now. Though, imo, this is not a good way to deal with it. PR addresses the "coping mechanism" for something that is fundamentally problematic

**Comment by utilbird** [comment]
Ground up rework isn't an easy task though. Making tendrils and Megafauna more difficult to beat would be a better solution for most of the changes in this PR.

**Comment by ghost** [comment]
no

**Comment by Rukofamicom** [comment]
> Making tendrils and Megafauna more difficult to beat would be a better solution for most of the changes in this PR.

That doesn't make the sheer power against other players available to miners any less of a problem. The issue I have with miners isn't how easy it is to obtain the gear, it's how powerful that gear is - there's already only a subset of players have the skills required to obtain the items and they deserve to be recognized for their ability to overcome megafauna... But that recognition should not come at the sacrifice of game balance. 

A player which is already extremely skilled and dangerous to other players does not need to have an overwhelming gear advantage to show off how good they are. Ash Drake armor shows off the fact they killed an Ash Drake just fine; they don't need the ability to cast wizard fire balls on top of it.

**Comment by Sinestia** [comment]
Orange man bad, mediocre PR

this review was sponsored

**Comment by Lagomorphica** [comment]
Well, I'm not a fan for a variety of reasons, but it is mostly minor complaints - I play shaft miner semi regularly, and while I'm not the best at fighting megafauna due to my 300 ping, I don't feel this is the best way to go about this. It might sound needy and dickish of me to incentivize changing instead of removing, but I still feel it'd be healthier compared to removing so much. I am also not good at porting or coding, so forgive me if I assume something about the ease of porting something.

### The Why

<details>
<summary> Why I am not a fan </summary>

Staff of Storms is only available from Legion, and aside from the chests - which are a random chance drop and have been nerfed. The wicker doll isn't exactly very useful, the hardsuits are good to have since its fire resistance - perhaps nerfing its station performance and making them more suited towards lavaland, such as the Champions Hardsuit, would make the most sense.

The katana deals 40 damage, yeah, but there are other implementations - making it cursed so you can't drop it is one I can think of, or making it give you a brain trauma akin to the Cursed Katana. I use it to fight against Goliaths but I'm aware that isn't a very common use for it.

 ~~It would also prevent my Bury The Light Vergil cosplay https://youtu.be/Jrg9KxGNeJY .~~

The Ash Drake at this point has little in good loot except for dragons blood and the armor, and well.. dragons blood is an RNG chance that two out of four times just makes you another race, sometimes a shitter one at that. Skeletons aren't good for mining, for instance, since your mining medipens no longer work, and if you get hit too hard you can lose a limb. Yeah, it has benefits, but you could also just.. turn in too a red digitigrade lizard and lose the ability to walk over shards. Lava staff and Sacred Flame were kind of garbage I admit, but Sacred Flame was not super overpowered since fire extinguishers exist that shut it down pretty fast, relative too the power of the ash drake suit they get instead. That doesn't mean nerf it though.

Anomalous Crystal is *literal garbage.* The only good one is Lightgeists, but even then, thats one effect out of a bunch of useless or shit ones. Making it a drop from Legion is even moreso stupid since its very rare for ash storm staffs to ever even come out in the first place, and they have utility. Ash storms only do 4 damage per second in a single room or in Space, and that.. is really not all that much. Or you can just wear fireproof clothing and be rather safe too it.

1500 points of ore is also very paltry and very insignificant - by the time people fight most megafauna, you can just plasmacut most of that in a very short amount of time, much shorter than compared to killing a large megafauna.

Instant Summons is more a utility than a combat spell so you could replace the Sacred Flame book with it I guess. It doesn't thematically fit, but oh well.

Holoparas in general are fucking garbage and shouldn't be an 18 TC traitor item but I can understand the removal, albeit they are exceptionally fun. If there was a way to tweak it so that its random that'd be cool, though.

Inferno grenade is just a clusterbang and its annoying as shit to get so yeah its nice to see it removed, as it has zero use outside of murder.

Cleaving saw nerf I'm kind of iffy on but I don't have enough experience to comment on the rest.

I can go on if you wish but removing a bunch of stuff does not sit very well with me - as most people say, typically when something is removed with the promises of something later, later never comes. Yes, I trust you to do it, but the question is if in that time something reasonable and interesting can be made up - not just adding the Diamond Pickaxe again, which also should probably be removed from chest loot.

</details>

### An Addendum On Security

<details
<summary> On Security and Antagonists </summary>

While I am aware that you believe Security is the strongest department, or should be, this is very flagrantly and blatantly untrue, even from a PvP perspective. The only real advantage they have is impunity to act over the crew and arrest people whereas for others it'd be considered powergaming or similar. They have zero protection from antagonists, and can be freely killed, along with the fact that the station very rarely sides with them. At the same time, however, Security buffs or similar are almost always met with criticism - as Security is at its strongest early on in the game. Atmospherics took a major nerf with the addition of fastmos, since the most available deadly weapons an atmos tech can deploy is a literal minor inconvenience at this point. A fusion death mix can be countered by just removing a single tile. 

But then you also have Botany, Cargo, Chemistry and Xenobiology, Science in general, Robotics, Genetics, etcetera. Engineering and Atmospherics are certainly weaker, but you can still make stunprods and similar, or singletank bomb people/make the doors shock people so hard it instahusks them.

Botany? Self explanatory, they can end the entire stations existence in mere moments. A lucky botanist or good botanist can get plasmaman mutation toxin in a decent timeframe and render any and all protection moot, or just get nitroglycerin plants. Hell, you could just get iron and uranium in a plant with sep. chems and you immediately render almost all of Securities non-lethal gear obcolescent and useless, which any Assistant could also do by making a mortar and pestle and grinding shit in maintenance.

Cargo? Order Russian Surplus crates, egun or shotgun crates, then print ammo. A Mosin Nagant does 60 brute damage. 60 brute damage. More than a point blank shotgun round, and equivalent to a shotgun slug. The mosin also fits in normal bags without being cut down at all. All of its ammo, 5 rounds, can be reloaded instantly and carried on stripper clips instead of individually, too! And it has more range. Maybe its a bit more RNG, sure, but you usually get Combat Gloves, an ammo crate, armor vests, etcetera. You can carry a LOT of mosin ammo, too.

Chemistry? Death chem mixes abound - you can make grenades that will pierce the toughest of hides, that will create plasma fire in space, and syringes that will go through all armor and all but ensure death to everybody. Sabotage telecommunications with a 50/50 EMP, or just completely neuter almost all Security officers to the point of uselessness since NT is energy weapon based.

Xenobiology has been covered to death and you already have plans to nerf it, but Science has not.

Science can teleport all the stuff out of the Armory, teleport maxcaps in too peoples departments and solo nuke them, which Security has no protection against, get strange devices that are flashbangs in hand and cooperate to get a free infinite use flashbang, use Robotics to get whatever the fuck they want, bag of holding bomb, bluespace body bag people and hide all evidence whilst also killing those unconscious players, and produce any other departments machines, as can Engineering.

Genetics is genetics and can fuck over every antag on station by just distributing hulk and X-Ray with zero repercussion or risk to themselves or anyone as they are zero downside mutations. There's no proper counterplay except.. what, shoot the guy with mutadone and tell everyone that you're shooting syringes of mutadone at people? And doing it to everyone, when you might not even be able to tell WHO has x-ray?

But there is also another issue - the fact that Security rarely uses the full extent of its tools available. Most people only ever use the shotguns or the eguns, and skip over the robustness of alternate ammo types - they do not make slugs, pulse slugs, laser scatter rounds, etcetera, all of which completely bypass almost everything being removed by virtue of their immense damage to anything it hits. Security consistently gets nerfed in too the ground for being overly strong, and are gimped and limitated in a variety of different ways that render them incapable of achieving the power they should have. The Detective can kill anyone in a firesuit or slow them, or track them, using .38 ICEBLOX or .38 TRAX, they can make laser slugs that 90 damage per direct hit, yes, 90 fucking damage, and you can load 6 in a shotgun over standard buckshot. .38 ICEBLOX is certain death in a hardsuit and a massive slow, as it sets your body temperature as low as possible. 

However - how often do you see Security use all their gear? Their ability to shoot people through walls with X-Ray lasers, to bola people, to spam flashbangs with a nade launcher or by throwing them, to request gear from other departments if they so wish, and to use their full ability to their advantage? Security can secure a great tactical advantage, but so rarely do so, simply because it takes input of effort for a department given a large amount of gear at the start.

This is also similar to antagonists - judging antagonists by the strength of the gear they get is quite poor. You can easily do more damage and deal more death out with station equipment - stun batons, for instance, are arguably the most powerful weapons on station. If you apply understanding towards the station on how to do things, especially in situations like Hijack, where you /can/ murderbone, then you will be infinitely more robust than somebody who has 9999 TC but doesn't bother to use any of their gear on station, solely buying things through the Uplink.

Do I believe miners are overtuned in some ways? Yes, certainly, in some ways they are, albeit I have almost never encountered a miner who has been uncounterable by anything at all, only that people haven't -thought- of those counters at the time. 

TL;DR, Security and Antagonists are already not the most powerful force on station. Generally its accepted that Security is strongest round-start, but loses to other departments as they scale, albeit this is also an issue of the fact people don't bother to fully utilize the gear they are given and instead rely on shotguns + buckshot or just raw energy guns.
</details>

**Comment by ToasterBan** [comment]
why voodoo dolls removal tho?

**Comment by Rukofamicom** [comment]
> why voodoo dolls removal tho?

To move it somewhere else - like curator/stage magician uplink. It's a quirky item that's mostly useless outside of antagonizing people, so it fits the bill for a quirky antag item.

**Comment by OceanFish1** [comment]
nerf??

**Comment by VictorVow** [comment]
> PR

Also change meat hook to use a knockdown not a stun please, this is should have removed ages ago like every other old stun was and this has been gone on other servers for like 2 years now. 

> However - how often do you see Security use all their gear?

Unlike other departments who can spend the whole shift powergaming then surface later with all their epic loot (not just referring to mining here), Security has a job which requires them to interact with crew and stay on the move most of the shift.

Thus you rarely see powergamed to the max weaponry, nor the Brig fortified to extreme levels. It's less fun for everyone to do so also, and besides people would just cry for even more nerfs saying "SHITCURITY TOO STRONK!" if they powergamed like other departments did.

**Comment by Lagomorphica** [comment]
> Unlike other departments who can spend the whole shift powergaming then surface later with all their epic loot (not just referring to mining here), Security has a job which requires them to interact with crew and stay on the move most of the shift.
> 
> Thus you rarely see powergamed to the max weaponry, nor the Brig fortified to extreme levels. It's less fun for everyone to do so also, and besides people would just cry for even more nerfs saying "SHITCURITY TOO STRONK!" if they powergamed like other departments did.
---

This is true and I did kind of say it in my post about people getting shitty when Security gets buffed. Shotguns get nerfed all the time, energy guns got nerfed with a size increase, etcetera. People don't want Security to be strong in any way, even if they don't bother to powergame to the max - walking up to an protolathe and printing shit is one example. I feel like buffing Security in some way or making them have an incentive or allowance to actually get gear from the protolathe would make more sense, and hopefully reduce the amount of people that just grab antag gear as Sec and run it on golden and sage.

**Comment by yyzsong** [comment]
> Also change meat hook to use a knockdown not a stun please,

Echoing this. The clown staff from the bananium ruin could also use this treatment, given that it's a ranged chainstun, honk.

**Comment by Rukofamicom** [comment]
Currently the meat hook utilizes paralyze rather than a full stun - it prevents movement but not action unless I'm mistaken in reading the code for paralyze.

**Comment by ItsTisme** [comment]
The chaplain hardsuit IMO was fun and should stay.

**Comment by pismorizzo** [comment]
Inconsistency in the loot is why I keep playing shaft miner, if I got the same items every round I would rarely play it as much as I have. You seek to "fix" mining by making every round as miner much closer together by removing 70% of the good tendril drops and 2/3s of the boss items. This is in no way a overhaul this is a nerf or more accurately a purge with no additions to replace the removals.

**Comment by ghost** [comment]
The main reason these lavaland nerf PRs keep getting closed is that they remove broken items without replacing them with anything else. Rewards are what keep miners playing and shrinking the loot pool makes it more boring, since there's nothing else to mining except fighting megafauna. I'd recommend only removing the most egregious loot drops (ERT suits, holoparasite (or remove their ability to infinitely heal you without even manifesting), these two are the worst things to deal with every round) and continuing with your rebalances.

Puzzle cube is badly coded and allows you to drop whichever cube you're holding before the teleport finishes, allowing you to keep teleporting back and forth. It's meant to be one-way, so consider fixing that to nerf it.

Fireball itself needs to be nerfed, rather than the wand. I thought this was done ages ago on TG but seems to not have made it here. High damage, knockdown and fire is too powerful regardless of what it comes from, be it a wizard or megafauna loot. It ganks any humanoid without fail.

Hierophant Staff damage nerf, good. Half the issue was that powergamers can just hide it in a bag of holding and never risk having it stolen/disarmed before a fight, where it then becomes an esword in melee, with the other half being that it lets you bypass defenses anyone puts up in a part of the station and just spam their location with AOE attacks.

Meat hook, again, the issue was mostly that powergamers can put it in their bag of holding and never risk losing it, then whip it out in a fight and get a free stun + knockdown + damage. I don't remember if it still does a full stun that blocks attacks, but during the knockdown you can still shove your target and have them stunned for 4 seconds, and kill them easily.

Cursed katana is also meant to have NODROP on it and give you the hunting spectre trauma as far as I remember, but nobody's bothered to fix it.

For the rest of the items you're removing, they need replacement instead. Griefing with them is something admins are meant to deal with. Thankyou for reading

**Comment by Rukofamicom** [comment]
> Puzzle cube is badly coded and allows you to drop whichever cube you're holding before the teleport finishes, allowing you to keep teleporting back and forth. It's meant to be one-way, so consider fixing that to nerf it.

You are confusing the red/blue cubes with the puzzle cube. The teleportation cubes are not being removed. The puzzle cube is an item you use on another player to trap them inside of a 3x3 sliding puzzle that frequently forms partially inside walls or machinery and can't be solved without disassembling close to a 5x5 space around it.

In many cases this translates to - completely removes a player from the round. 

> Cursed katana is also meant to have NODROP

It does have this. This makes the item worse for a variety of reasons.
* You can't disarm the player that has it, making it especially powerful in combat situations
* Miner that picks it up but doesn't want to valid-hunt is now gimped somewhat heavily depending on their preferred mining style, until they get their arm surgically removed and re-attached.
* It is just a totally useless drop to the miner that doesn't want to run away and fight players, but is aware they can't drop it if they touch it.

All in all it's an awful "reward" for miners any way you look at it. 

> Griefing with them is something admins are meant to deal with. 

Hi, I am an admin and I am dealing with the grief they cause :smile:

In all seriousness though - items with no purpose outside of griefing don't have a place on Beestation anymore. An item that can't even be conceivably used by antagonists under current rules specifically doesn't have a place.

**Comment by Rukofamicom** [comment]
Misclicked on phone 😔

**Comment by Rukofamicom** [comment]
> this is a nerf or more accurately a purge with no additions to replace the removals.

Removal of some items makes it more likely to get others. Items removed were either trash drops nobody ever wanted, or stuff that's too powerful to be widely available to non-antagonist players just because they got lucky and are able to fight bosses.

Good loot as well as fun loot is still available. You will always know what you will get from specific bosses and you still won't know what you'll get from a tendril, but it's less likely to be total garbage as well as less likely to be completely busted.

But yes, at it's core you are correct - this is intended to be a heavy nerf to what shaft miners can achieve in terms of maximum power level. If you take a moment to go over what's still available, even after the nerfs, they are still going to be extremely strong - probably still the strongest overall despite being weaker than they were. **Unless you disagree with this statement and back it up with proof, the loot removed does not need to exist**

Come up with and suggest fun loot that isn't also extremely powerful, it will pass in a heartbeat.

**Comment by Sinestia** [comment]
> 
> 
> Misclicked on phone 😔

tricking all of the gamers w/ email notifications thinking you gave up on this lmao

**Comment by Rukofamicom** [comment]
> Also change meat hook to use a knockdown not a stun please

> Echoing this.

Swapped from paralyze (which isn't a full stun?) to just a knockdown.

**Comment by pismorizzo** [comment]
The main item I am upset is going is specifically the champion hardsuit as seeing miners walk around in it is a huge part of mining flavor, just add the champion suit to the Legion boss drops as the Legion is way too annoying to fight over just 1-3 chests and the crystal  (which has no use other than to mess around with) and almost nobody fights the Legion anyhow. Hell if you feel the suit itself is too strong even for a boss drop just remove the magic immunity or lower resistances so it is weaker than the ash drake suit but with space proof and immune to magic.

**Comment by utilbird** [comment]
> as seeing miners walk around in it is a huge part of mining flavor

I'm sure miners and the rest of the crew will find a way to bare without their ERT hardsuits.

**Comment by jlsnow301** [comment]
> this allows players who find normal mining boring to still acquire ores that help the station as a whole as a reward for completing boss fights

Incredibly based

**Review by utilbird — APPROVED** [review]
Meh. Mining is broken fundamentally. While not a perfect fix, this should reduce antag suffrage from post-mining validhunters for the time being. So that's a tick from me

**Comment by Rukofamicom** [comment]
Sadly a bandage is the best I can manage for now. A full and proper overhaul of mining as a whole would be a pretty sizeable brainstorming session and equally sizable bounty away. *It would likely still be hated since the over the top shit would still be removed from it anyway*.

To any ideasguys out there though, come with alternative mining systems that would follow a progression system of upgrades and optional boss fights, without rewarding non-antags who try to abandon their obligations and preferably making the actual mining process more enjoyable. *I'll slap a bounty on good ideas that meet this criteria*.

**Comment by Rukofamicom** [comment]
Sadly a bandage is the best I can manage for now. A full and proper overhaul of mining as a whole would be a pretty sizeable brainstorming session and equally sizable bounty away. *It would likely still be hated since the over the top shit would still be removed from it anyway*.

To any ideasguys out there though, come with alternative mining systems that would follow a progression system of upgrades and optional boss fights, without rewarding non-antags who try to abandon their obligations and preferably making the actual mining process more enjoyable. *I'll slap a bounty on good ideas that meet this criteria*.
